packages/kubernetes-1.20/credential-provider-config-yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: kubelet.config.k8s.io/v1alpha1
+kind: CredentialProviderConfig
+providers:
+{{#if settings.kubernetes.credential-providers}}
+{{#each settings.kubernetes.credential-providers.providers}}
+{{#if this.enabled}}
+  - name: {{this.name}}
+    matchImages:
+{{#each this.image-patterns}}
+      - "{{this}}"
+{{/each}}
+    defaultCacheDuration: "{{default "12h" this.cache-duration}}"
+    apiVersion: credentialprovider.kubelet.k8s.io/v1alpha1
+{{#if (eq this.name "ecr-credential-provider")}}
+    env:
+      - name: PATH
+        value: /bin:/usr/bin:/usr/local/bin
+      - name: HOME
+        value: /root
+      - name: AWS_SHARED_CREDENTIALS_FILE
+        value: /etc/kubernetes/kubelet/.aws/config
```

Comment:
`/etc/aws/config` is probably a more standard way to render this. Since the file could have secrets, we would also want to ensure that the directory is mounted with the `secret_t` label, like `/etc/containerd` is.

Reply:
Capturing here - Ben and I discussed this and decided to move the config to the more common location of the home directory. So latest version moves things to `/root/.aws/config`.